src/js/BookReader/BookModel.js

```diff
@@ -272,18 +322,81 @@ export class BookModel {
    * Helper. Return a prop for a given index
    * @param {PageIndex} index
    * @param {keyof PageData} prop
+   * @param {*} def default if property not on the record
    * @return {*}
    */
-  _getDataProp(index, prop) {
+  _getDataProp(index, prop, def=undefined) {
     const dataf = this._getDataFlattened();
     if (isNaN(index) || index < 0 || index >= dataf.length)
       return;
     if ('undefined' == typeof(dataf[index][prop]))
-      return;
+      return def;
     return dataf[index][prop];
   }
 }
 
+/**
+ * A controlled schema for page data.
+ */
+class BookPage {
```

Comment:
:+1: Will rename to `PageModel`

I'd rather leave PageModel here for now; it's tightly coupled to BookModel since it requires a BookModel to be constructed.

Reply:
Renamed :+1: